File: sort/sort.rb
arr = [
  {:name => :a, :score => 10},
  {:name => :b, :score => 5},
  {:name => :c, :score => 7},
]

p arr.sort_by { |item| item[:score] }

p arr.sort { |a, b| a[:score] <=> b[:score] }

File: sort/sort.pl
use 5.016;
use strict;
use warnings;

use Data::Dumper;

my $arr = [
    +{
        name => 'a',
        score => 10
    },
    +{
        name => 'b',
        score => 5
    },
    +{
        name => 'c',
        score => 7
    },
];

print Dumper [sort {$a->{score} <=> $b->{score}} @$arr];
